Task4/src/main/java/by/innowise/tasks/service/DepartmentService.java

```diff
@@ -0,0 +1,57 @@
+package by.innowise.tasks.service;
+
+import by.innowise.tasks.dto.DepartmentDto;
+import by.innowise.tasks.handler.NotFoundException;
+import by.innowise.tasks.mapper.DepartmentMapper;
+import by.innowise.tasks.repository.DepartmentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class DepartmentService {
+
+    public static final String NOT_FOUND_BY_ID = "There is no such department with id: %s";
+
+    @Autowired
+    private DepartmentRepository departmentRepository;
+
+    @Autowired
+    private DepartmentMapper departmentMapper;
+
+    public DepartmentDto saveDepartment(DepartmentDto departmentDto) {
+        return departmentMapper
+                .toDepartmentDto(departmentRepository
+                        .save(departmentMapper
+                                .toDepartment(departmentDto)));
+    }
+
+    public List<DepartmentDto> getAllDepartments() {
+        return departmentRepository
+                .findAll()
+                .stream()
+                .map(departmentMapper::toDepartmentDto)
+                .toList();
+    }
+
+    public DepartmentDto getDepartmentById(Long id) {
+        return departmentMapper
```

Comment:
Instead of mapping result of optional calls we can use method map inside optional